src/Compilers/CSharp/Test/Emit2/Semantics/RefStructInterfacesTests.cs

```diff
@@ -0,0 +1,2914 @@
+                //         [UnscopedRef]
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "UnscopedRef").WithArguments("ref struct interfaces").WithLocation(8, 10)
+                );
+        }
+
+        [Fact]
+        public void UnscopedRefInInterface_Property_06()
+        {
+            var src = @"
+using System.Diagnostics.CodeAnalysis;
+
+public interface I
+{
+    ref int P
+    {
+        [UnscopedRef]
+        get
+        {
+            throw null;
+        }
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80);
+
+            CompileAndVerify(comp, sourceSymbolValidator: verify, symbolValidator: verify, verify: Verification.Skipped).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                PropertySymbol propertySymbol = m.GlobalNamespace.GetMember<PropertySymbol>("I.P");
+                Assert.False(propertySymbol.HasUnscopedRefAttribute);
+                Assert.True(propertySymbol.GetMethod.HasUnscopedRefAttribute);
+            }
+
+            CreateCompilation(src, targetFramework: TargetFramework.Net80, parseOptions: TestOptions.RegularNext).VerifyDiagnostics();
+
+            CreateCompilation(src, targetFramework: TargetFramework.Net80, parseOptions: TestOptions.Regular12).VerifyDiagnostics(
+                // (8,10): error CS8652: The feature 'ref struct interfaces' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         [UnscopedRef]
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "UnscopedRef").WithArguments("ref struct interfaces").WithLocation(8, 10)
+                );
+        }
+
+        [Fact]
+        public void UnscopedRefInInterface_Property_07()
+        {
+            var src = @"
+using System.Diagnostics.CodeAnalysis;
+
+public interface I
+{
+    sealed ref int P
+    {
+        [UnscopedRef]
+        get
+        {
+            throw null;
+        }
+    }
+}
+";
+            var comp = CreateCompilation(src, targetFramework: TargetFramework.Net80);
+
+            comp.VerifyDiagnostics(
+                // (8,10): error CS9101: UnscopedRefAttribute can only be applied to struct or virtual interface instance methods and properties, and cannot be applied to constructors or init-only members.
+                //         [UnscopedRef]
+                Diagnostic(ErrorCode.ERR_UnscopedRefAttributeUnsupportedMemberTarget, "UnscopedRef").WithLocation(8, 10)
+                );
+
+            PropertySymbol propertySymbol = comp.GetMember<PropertySymbol>("I.P");
+            Assert.False(propertySymbol.HasUnscopedRefAttribute);
+            Assert.False(propertySymbol.GetMethod.HasUnscopedRefAttribute);
+        }
+
+        [Fact]
+        public void UnscopedRefInInterface_Property_08()
```

Comment:
Consider testing also indexers.

Reply:
> Consider testing also indexers.

While I can certainly clone property tests with indexers, I do not think the tests will have any immediate value because from the symbol model point of view indexers are properties, they do not have separate representation or handling.